### PR TITLE
IT: Simplify liquidity_pool test module

### PR DESCRIPTION
**Pull request by lemunozm**
# Description

There are a lot of overlapping and boilerplate in the current liquidity-pool tests.

- Minor IT utilities improvements & simplifications.
- Added an `approx()` to check approximation amounts.
- Give more "multi-runtime" support in several tests
- Extracted and refactored some modules from `liquidity_pools.rs` into their own files, trying to avoid as much testing overlapping as possible:
  - Module `transfer` into `xcm_transfer.rs` file:
    - [x] para -> para
    - [x] relay -> para -> relay
  - Module `asset_registry` into `assets.rs` file.
  - Module `currency_id_convert` into `currency_conversion.rs` file.
  - Module `restricted_tokens` into the already existent `restricted_tokens.rs` file.
- Remove fudge from proxy tests.
- `PARA_ID` and `SIBLING_ID` no longer tied to `FudgeHandle`. They are gloval values that can be read in any place without using `T::FudgeHandle`
  
**I do not touch any tests related to liquidity pools itself.** I want to give them support for altair and centrifuge and remove the Fudge requirement in another PR. Also, I think **there is a lot of overlap with the new `lp` module**, so maybe some of them can just be removed.

### Next in line (future PRs):
- Move gateway tests to its own module tested for all runtimes: https://github.com/centrifuge/centrifuge-chain/pull/1879
- Move routers tests to its own module tested for all runtimes.
- liquidity pool tests without fudge, and tested for all runtimes.

### CODE REVIEW

File: runtime/integration-tests/src/utils/mod.rs [hunk 1]
```diff
@@ -12,3 +12,73 @@
...
+
+	#[test]
+	fn approximations() {
+		assert_eq!(1000u128, 996.approx(-0.01));
+		assert_eq!(1000u128, 1004.approx(0.01));
+		assert_eq!(1000u128, 1500.approx(0.5));
+
+		assert_ne!(1000u128, 996.approx(0.01));
+		assert_ne!(1000u128, 1004.approx(-0.01));
+		assert_ne!(1000u128, 1500.approx(0.1));
```
[the remark this answers is not in the document]
Comment (lemunozm) [reply runtime/integration-tests/src/utils/mod.rs:82]: The value `0.01` means that both values are equal if the right-hand side is as much as 1% higher than the other one. `-0.01` means lower than.

**Comment by lemunozm** [comment]
There are a lot of git diffs in `liquidity_pools.rs` file. But 99% of the changes is just removing the modules `transfer`, `asset_registry` and `currency_id_convert`

File: runtime/integration-tests/src/generic/cases/currency_conversions.rs [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+			.storage(),
+	);
+
+	env.parachain_state(|| {
+		assert_eq!(
+			CurrencyIdConvert::<T>::convert(Location::new(0, Here)),
+			Some(curr.id()),
+		);
+
+		assert_eq!(CurrencyIdConvert::<T>::convert(curr.id()), None);
```
Comment (lemunozm) [inline runtime/integration-tests/src/generic/cases/currency_conversions.rs:86]: BTW, here I would expect a symmetric behavior, but it isn't.


The implementation only checks the transferability when converting from `location -> currency` [ref](https://github.com/centrifuge/centrifuge-chain/blob/main/runtime/common/src/xcm.rs#L175). Should we add the same filter for `currency -> location` ?

File: libs/types/src/tokens.rs [hunk 3]
```diff
@@ -364,6 +364,23 @@ impl CrossChainTransferability {
...
+	}
+
+	/// Fees will be charged using `AssetRegistryTrader`.
+	/// If value is 0, no fees will be charged.
+	#[cfg(feature = "std")]
+	pub fn xcm_with_fees(value: Balance) -> Self {
+		Self::Xcm(XcmMetadata {
+			fee_per_second: Some(value),
+		})
+	}
```
Comment (lemunozm) [inline libs/types/src/tokens.rs:383]: Tech-debt to create some tests to be able to compute the exact fee charged in the above two cases. Using `FixedRateOfFungible` and using `AssetRegistryTrader`

**Review by cdamian — APPROVED** [review]

**Review by wischli — DISMISSED** [review]
Super organized and clean, love it! Two minor nits, which can be ignored and one comment from my side.

File: runtime/integration-tests/src/generic/cases/xcm_transfers.rs [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+			},
+		},
+	},
+	utils::{accounts::Keyring, approx::Approximate},
+};
+
+const INITIAL: u32 = 100;
+const TRANSFER: u32 = 20;
+
+fn create_transfeable_currency(decimals: u32, para_id: Option<u32>) -> CustomCurrency {
```
[the remark this answers is not in the document]
Comment (lemunozm) [reply runtime/integration-tests/src/generic/cases/xcm_transfers.rs:31]: Eagle eye! Thanks

**Comment by lemunozm** [comment]
@wischli @cdamian can you re-approve this? 🙏🏻

**Review by cdamian — APPROVED** [review]

**Pull request merged**